flask/app.py

```diff
@@ -525,10 +531,14 @@ def __init__(self, import_name, static_path=None, static_url_path=None,
         # while the server is running (usually happens during development)
```

Comment:
`has_static_folder` doesn't indicate whether the folder exists or not, only if one has been configured.